### PR TITLE
ListView should convert item to string when ItemTemplate is not specified

### LINKED ISSUE #3715: [Feedback] Xamarin UWP ListView not calling ToString()
### Description

<p>Given a Xamarin.XAML ContentPage definition containing a ListView component name &quot;entries&quot;, the following will display...

entries.ItemsSource =  new List&lt;string&gt; { Guid.NewGuid().ToString(), Guid.NewGuid().ToString() };

The following will not display:

entries.ItemsSource =  new List&lt;Guid&gt; { Guid.NewGuid(), Guid.NewGuid() };</p>
<a href="https://developercommunity.visualstudio.com/content/problem/315382/xamarin-uwp-listview-not-calling-tostring.html">View on Developer Community</a>
<br>To find diagnostic information, see the <a href="https://devdiv.visualstudio.com/DevDiv/_workitems/edit/666803">original linked feedback ticket</a>.

### Source

[677729](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/677729)

### PR DESCRIPTION
**Pull request by duracellko**
### Description of Change ###

Makes behavior of **ListView** without **ItemTemplate** compatible with Xamarin.Forms version 2.x. Behavior of **ListView** with **ItemTemplate** specified is not changed.

Xamarin.Forms version 2.x used directly `ToString` method on each item, when `ItemTemplate` was not specified. Version 3.x changed from direct `ToString` call to data binding. This works fine, when item is string. However, binding results to empty cell (with null value), when item is not string type. Therefore it introduces incompatibility between Xamarin.Forms version 2 and 3.

This change introduces `ToStringValueConverter` that converts any object to string using `object.ToString` or `IFormattable.ToString`. This value converter is then used with data binding in `ListView` without `ItemTemplate`.

### Issues Resolved ### 

- fixes #3715 

### API Changes ###

Added:
 - class ToStringValueConverter : IValueConverter

### Platforms Affected ### 

- Core/XAML (all platforms)

### Behavioral/Visual Changes ###

This update makes `ListView` without `ItemTemplate` behavior compatible with Xamarin.Forms version 2.x. It means that string values are displayed, when `ItemsSource` contains other objects than strings.

### Before/After Screenshots ### 

Not applicable

### Testing Procedure ###

Added unit tests to cover the changes. Integration tests were not modified.

### PR Checklist ###

- [x] Has automated tests (unit-tests only)
- [x] Rebased on top of the target branch at time of PR
- [x] Changes adhere to coding standard

### CODE REVIEW

**Comment by kingces95** [comment]
Hm, I'm getting a build error in your test. 
image

**Comment by kingces95** [comment]
image

**Comment by PureWeen** [comment]
Failing Android test unrelated

File: Xamarin.Forms.Core/ToStringValueConverter.cs [hunk 1]
```diff
@@ -0,0 +1,28 @@
+﻿using System;
+using System.Globalization;
+
+namespace Xamarin.Forms
+{
+	public class ToStringValueConverter : IValueConverter
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Core/ToStringValueConverter.cs]: should have been made internal

**Pull request merged**